src/Standards/Generic/Tests/CodeAnalysis/UselessOverridingMethodUnitTest.1.inc

```diff
@@ -76,4 +87,56 @@ interface InterfaceFoo {
 
 trait TraitFoo {
     abstract public function sniffShouldBailEarly();
+
+    public function usefulMethodInTrait() {
+        parent::usefulMethodInTrait();
+
+        return 1;
+    }
+
+    public function uselessMethodInTrait() {
+        return parent::uselessMethodInTrait();
+    }
+}
+
+abstract class AbstractBar extends AbstractFoo {
+    public function uselessMethodInAbstractClass() {
+        parent::uselessMethodInAbstractClass();
+    }
+
+    public function usefulMethodInAbstractClass() {
+        $a = 1;
+        parent::usefulMethodInAbstractClass($a);
+    }
+}
+
+enum EnumFoo {
+    public function sniffShouldBailEarly() {
+        // Invalid code needed to ensure an error is NOT triggered and the sniff bails early when handling an enum method.
+        parent::sniffShouldBailEarly();
+    }
+}
+
+function shouldBailEarly() {
+    // Invalid code needed to ensure an error is NOT triggered and the sniff bails early when handling a regular function.
+    parent::shouldBailEarly();
+}
+
+new class extends ParentClass {
```

Comment:
```suggestion
$anon = new class extends ParentClass {
```

Reply:
I just added this in a new commit, but I wanted to make sure I understand why you suggested this. Is it to highlight that this test is for anonymous classes or something else that I'm missing?

Reply:
It's mostly to make it realistic code. Without assigning the class to a variable, there would be no way to use the anonymous class.